src/service/api.tsx

```diff
@@ -124,6 +128,50 @@ const api = {
     return obj.data.results;
   },
 
+  async getSearchFileredList(dataFilter:
+    {
+      from: number,
+      to: number,
+      gener: string | null,
+      page: number
+    } = {
+      from: 1900,
+      to: 2020,
+      gener: null,
+      page: 1,
+    }
+  ) {
+
+    function setFilteredData() {
+      const { from, to } = dataFilter
+      if (from < to) {
+        return `&primary_release_date.gte=${from}-01-01&primary_release_date.lte=${to}-01-01`
+      } else if (from > to) {
+        return `&primary_release_date.gte=${to}-01-01&primary_release_date.lte=${from}-01-01`
+      }
+      return `&primary_release_date.gte=${from}-01-01`
+    }
+
+    function setGener() {
+      const { gener } = dataFilter;
+      if (gener !== null) {
+        const generId = API_GENER_ID.find(item => item.name === gener);
+        return `&with_genres=${generId?.id}`
```

Comment:
can be &with_genres=undefined with this implementation

Reply:
no, we have convention to use only known genres for us